MatrixSDKTests/MXRoomTests.m

```diff
@@ -516,6 +518,113 @@ - (void)tesRoomDirectoryVisibilityLive
+    NSString *thirdMessageFormattedReplyToSecond = @"<p><strong>Reply to second message</strong></p>";
+    
+    NSString *expectedThirdEventBodyStringFormat = @"> <%@> **Reply to first message**\n\n**Reply to second message**";
+    NSString *expectedThirdEventFormattedBodyStringFormat = @"<mx-reply><blockquote><a href=\"%@\">In reply to</a> <a href=\"%@\">%@</a><br><p><strong>Reply to first message</strong></p></blockquote></mx-reply><p><strong>Reply to second message</strong></p>";
+    
+    MXSendReplyEventDefaultStringLocalizations *defaultStringLocalizations = [MXSendReplyEventDefaultStringLocalizations new];
+    
+    [matrixSDKTestsData doMXSessionTestWithBobAndARoomWithMessages:self readyToTest:^(MXSession *mxSession, MXRoom *room, XCTestExpectation *expectation) {
+        
+        __block NSUInteger messageCount = 0;
+        
+        // Listen to messages
+        [room.liveTimeline listenToEventsOfTypes:@[kMXEventTypeStringRoomMessage] onEvent:^(MXEvent *event, MXTimelineDirection direction, MXRoomState *roomState) {
+            messageCount++;
+            
+            if (messageCount == 1)
+            {
+                __block MXEvent *localEchoEvent = nil;
+                
+                // Reply to first message
+                [room sendReplyToEvent:event withTextMessage:secondMessageReplyToFirst formattedTextMessage:secondMessageFormattedReplyToFirst stringLocalizations:defaultStringLocalizations localEcho:&localEchoEvent success:^(NSString *eventId) {
+                    NSLog(@"Send reply to first message with success");
+                } failure:^(NSError *error) {
+                    XCTFail(@"The request should not fail - NSError: %@", error);
+                    [expectation fulfill];
+                }];
+                
+                XCTAssertNotNil(localEchoEvent);
+                
+                NSString *roomId = room.roomId;
+                NSString *firstEventId = event.eventId;
+                NSString *firstEventSender = event.sender;
+                
+                NSString *secondEventBody = localEchoEvent.content[@"body"];
+                NSString *secondEventFormattedBody = localEchoEvent.content[@"formatted_body"];
+                NSString *secondEventRelatesToEventId = localEchoEvent.content[@"m.relates_to"][@"m.in_reply_to"][@"event_id"];
+
+                NSString *permalinkToUser = [MXTools permalinkToUserWithUserId:firstEventSender];
+                NSString *permalinkToEvent = [MXTools permalinkToEvent:firstEventId inRoom:roomId];
+
+                NSString *expectedSecondEventBody = [NSString stringWithFormat:expectedSecondEventBodyStringFormat, firstEventSender];
+                NSString *expectedSecondEventFormattedBody = [NSString stringWithFormat:expectedSecondEventFormattedBodyStringFormat, permalinkToEvent, permalinkToUser, firstEventSender];
+
+                XCTAssert([secondEventBody isEqualToString:expectedSecondEventBody]);
```

Comment:
Use XCTAssertEqualObjects to compare  strings. This is helpful to debug remotely.